File: java/sqlbuilder/Resource.java
package com.clickability.cms.dataaccess.sqlbuilder;

public class Resource implements Joinable {

	public final String resource;
	public final String alias;

	public Resource(String resource) {
		this(resource, null);
	}
	
	public Resource(String resource, String alias) {
		this.resource = resource;
		this.alias = alias;
	}
	
	public String alias() {
		return this.alias==null ? resource : alias;
	}
	
	@Override
	public String get() {
		return this.alias == null ? 
			this.resource : 
			String.format("%s as %s", this.resource, this.alias);
	}

}

File: src/main/kotlin/queryBuilder/com/inunison/lib/queryBuilder/Resource.kt
package com.clickability.cms.dataaccess.sqlbuilder

class Resource @JvmOverloads constructor(val resource: String, val alias: String? = null) : Joinable {

    fun alias(): String {
        return if (this.alias == null) resource else alias
    }

    @Override
    fun get(): String {
        return if (this.alias == null)
            this.resource
        else
            String.format("%s as %s", this.resource, this.alias)
    }

}
